src/inmanta/ast/statements/generator.py

```diff
@@ -698,7 +697,8 @@ def execute(self, requires: Dict[object, object], resolver: Resolver, queue: Que
 
         # check if the instance already exists in the index (if there is one)
         instances: List[Instance] = []
-        lookup_table: list[abc.Sequence[tuple[str, object]]] = []
+        # register any potential index collision
+        collisions: MutableMapping[tuple[str, ...], Instance] = {}
```

Comment:
As far as I know there's little reason to prefer `MutableMapping` over `abc`, it's mainly relevant when you don't intend to mutate it. But I have no objections either, this is more of an FYI.